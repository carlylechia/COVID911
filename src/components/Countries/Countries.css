#countries-container {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(10rem, 1.5fr));
  background: linear-gradient(to bottom, rgb(222, 49, 99), #d70040);
  overflow-x: hidden;
}

.country-card:hover {
  z-index: 1;
}

#countries-container > .country-card:nth-child(even) {
  background: rgb(222, 49, 99);
  outline: 2px solid #d70040;
}

#countries-container > .country-card:nth-child(odd) {
  background: #d70040;
  outline: 2px solid rgba(238, 50, 81, 0.916);
}

.country-names {
  font-size: 1.5rem;
  color: #fff;
}
