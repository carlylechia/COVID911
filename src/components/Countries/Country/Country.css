#country-card {
  display: grid;
  grid-template-columns: 30% 70%;
  grid-template-rows: 50% 1fr 1fr;
}

#country-map {
  justify-self: center;
}

#country-name {
  align-self: end;
}

#new-cases {
  grid-row: 2/2;
  grid-column: 2/2;
}

#total {
  grid-row: 3/3;
  grid-column: 2/2;
}
