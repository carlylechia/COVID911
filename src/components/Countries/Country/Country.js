import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const Country = ({
  name, image, total, newCases, pageUrl,
}) => (
  <NavLink
    to={`${pageUrl}`}
    className="country-card"
    style={{ minHeight: '25vh' }}
  >
    <div
      style={{
        width: '100%',
        height: '70%',
        background: ` url(${image}) no-repeat 50% 50%`,
        backgroundSize: '50% 80%',
        opacity: '0.2',
      }}
    />
    <p className="country-names" style={{ marginLeft: '100%' }}>{name}</p>
    <span className="">
      Total:
      {` ${total}`}
    </span>
    <span className="">
      New Cases:
      {` ${newCases}`}
    </span>
  </NavLink>
);
Country.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  total: PropTypes.number.isRequired,
  newCases: PropTypes.number.isRequired,
  pageUrl: PropTypes.string.isRequired,
};

export default Country;
