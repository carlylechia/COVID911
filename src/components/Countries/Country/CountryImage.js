import { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

export const getCountrySlug = (name) => {
  let countryName = name.replace(/[_]/g, '-').replace('*', '').replace(',', '');
  if (countryName === 'diamond-princess' || countryName === 'ms-zaandam') countryName = 'ship';
  if (countryName === 'south-sudan') countryName = 'sudan';
  if (countryName.includes('congo')) countryName = 'congo';
  if (countryName === 'west-bank-and-gaza') countryName = 'palestine';
  if (countryName.includes('timor')) countryName = 'timor';
  if (countryName === 'us') countryName = 'usa';

  return countryName;
};

export const useCountryImage = (src) => {
  const [url, setUrl] = useState();

  useEffect(() => {
    if (!url) return;
    const img = new Image();
    img.onerror = () => {
      setUrl(`https://mapsvg.com/static/maps/geo-calibrated/${getCountrySlug(src)}.svg`);
    };
    img.src = url;
  }, [url]);

  useEffect(() => {
    setUrl(src ? `/images/${getCountrySlug(src)}.png` : '');
  }, [src]);

  return url;
};

const CountryImage = ({ src, alt, className }) => {
  const url = useCountryImage(src);
  return <img src={url} alt={alt} className={className} />;
};

CountryImage.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  className: PropTypes.string,
};

CountryImage.defaultProps = {
  className: '',
};

export default CountryImage;
