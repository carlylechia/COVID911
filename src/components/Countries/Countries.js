import { useSelector, shallowEqual } from 'react-redux';
import Country from './Country/Country';
import loading from '../../Images/loading.gif';
import './Countries.css';

const Countries = () => {
  const countryList = useSelector(
    (state) => (state.countriesReducer.filteredCountries ? state.countriesReducer.filteredCountries
      : state.countriesReducer.countries), shallowEqual,
  );
  const wait = useSelector((state) => state.countriesReducer.wait, shallowEqual);

  return (
    wait ? <img className="db center" width="100" height="100" style={{ margin: '2rem 47%' }} alt="loading" src={loading} />
      : (
        <>
          <h2
            className="white f6 tracked fw3 pa2"
            style={{
              background: 'rgb(220, 59, 79)', padding: '7px', color: 'silver', textAlign: 'center',
            }}
          >
            STATS BY COUNTRY
          </h2>
          <main id="countries-container">
            {countryList.map((country) => (
              <Country
                key={country.id}
                pageUrl={country.id}
                image={country.image}
                name={country.name}
                total={country.total}
                newCases={country.newCases}
              />
            ))}
            {!wait && !countryList.length
              && <p className="">No Matching Results</p>}
          </main>
        </>
      )
  );
};

export default Countries;
