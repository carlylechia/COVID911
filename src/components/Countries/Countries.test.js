import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../Redux/createStore';
import Country from './Country/Country';
import Countries from './Countries';

describe('Countries component test', () => {
  test('Countries snapshot', () => {
    const props = {
      name: 'country 1',
      image: 'country 1 image',
      total: 100,
      newCases: 50,
      pageUrl: 'url',
    };
    const {
      name, image, total, newCases, pageUrl,
    } = props;
    render(
      <Router>
        <Provider store={store}>
          <Country
            name={name}
            image={image}
            total={total}
            newCases={newCases}
            pageUrl={pageUrl}
          />
        </Provider>
      </Router>,
    );
    expect(render(
      <Router>
        <Provider store={store}>
          <Countries />
        </Provider>
      </Router>,
    )).toMatchSnapshot();
  });
});
