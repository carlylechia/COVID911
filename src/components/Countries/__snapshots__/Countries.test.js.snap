// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries component test Countries snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="country-card grow pointer flex flex-column justify-center items-center"
        href="/url"
        style="min-height: 25vh;"
      >
        <div
          style="color: rgb(255, 255, 255); position: absolute; right: 0px;"
          to="url"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm244.5 0l-81.9-81.1c-7.5-7.5-7.5-19.8 0-27.3s19.8-7.5 27.3 0l95.4 95.7c7.3 7.3 7.5 19.1.6 26.6l-94 94.3c-3.8 3.8-8.7 5.7-13.7 5.7-4.9 0-9.9-1.9-13.6-5.6-7.5-7.5-7.6-19.7 0-27.3l79.9-81z"
            />
          </svg>
        </div>
        <div
          class="cover"
          loading="lazy"
          style="width: 10rem; height: 50%; background-image: url(undefined); opacity: 0.2;"
        />
        <p
          class="pr3 white f4 b"
          style="font-size: 1.3rem; color: aliceblue; text-align: center;"
        >
          country 1
        </p>
        <p
          class="db pr3 white f5 b"
          style="font-size: 1rem; color: azure; text-align: center;"
        >
          Total:
           100
        </p>
        <p
          class="db pr3 white f5 b"
          style="font-size: 1rem; color: azure; text-align: center;"
        >
          New Cases:
           50
        </p>
      </a>
    </div>
    <div>
      <h2
        class="white f6 tracked fw3 pa2"
        style="background: rgb(220, 59, 79); padding: 7px; color: silver; text-align: center;"
      >
        STATS BY COUNTRY
      </h2>
      <main
        id="countries-container"
      >
        <p
          style="font-size: 1.5rem; color: silver; margin: 2rem auto; text-align: center;"
        >
          No Matching Results
        </p>
      </main>
    </div>
  </body>,
  "container": <div>
    <h2
      class="white f6 tracked fw3 pa2"
      style="background: rgb(220, 59, 79); padding: 7px; color: silver; text-align: center;"
    >
      STATS BY COUNTRY
    </h2>
    <main
      id="countries-container"
    >
      <p
        style="font-size: 1.5rem; color: silver; margin: 2rem auto; text-align: center;"
      >
        No Matching Results
      </p>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
